src/positioner/src/Positioner.js

```diff
@@ -34,236 +29,230 @@ const getCSS = ({ initialScale, animationDuration }) => ({
   }
 })
 
-export default class Positioner extends PureComponent {
-  static propTypes = {
-    /**
-     * The position the element that is being positioned is on.
-     * Smart positioning might override this.
-     */
-    position: PropTypes.oneOf([
-      Position.TOP,
-      Position.TOP_LEFT,
-      Position.TOP_RIGHT,
-      Position.BOTTOM,
-      Position.BOTTOM_LEFT,
-      Position.BOTTOM_RIGHT,
-      Position.LEFT,
-      Position.RIGHT
-    ]).isRequired,
-
-    /**
-     * When true, show the element being positioned.
-     */
-    isShown: PropTypes.bool,
-
-    /**
-     * Function that returns the element being positioned.
-     */
-    children: PropTypes.func.isRequired,
-
-    /**
-     * Function that returns the ref of the element being positioned.
-     */
-    innerRef: PropTypes.func.isRequired,
-
-    /**
-     * The minimum distance from the body to the element being positioned.
-     */
-    bodyOffset: PropTypes.number.isRequired,
-
-    /**
-     * The minimum distance from the target to the element being positioned.
-     */
-    targetOffset: PropTypes.number.isRequired,
-
-    /**
-     * Function that should return a node for the target.
-     * ({ getRef: () -> Ref, isShown: Bool }) -> React Node
-     */
-    target: PropTypes.func.isRequired,
-
-    /**
-     * Initial scale of the element being positioned.
-     */
-    initialScale: PropTypes.number.isRequired,
-
-    /**
-     * Duration of the animation.
-     */
-    animationDuration: PropTypes.number.isRequired,
-
-    /**
-     * Function that will be called when the exit transition is complete.
-     */
-    onCloseComplete: PropTypes.func.isRequired,
-
-    /**
-     * Function that will be called when the enter transition is complete.
-     */
-    onOpenComplete: PropTypes.func.isRequired
-  }
-
-  static defaultProps = {
-    position: Position.BOTTOM,
-    bodyOffset: 6,
-    targetOffset: 6,
-    initialScale: 0.9,
-    animationDuration: 300,
-    innerRef: () => {},
-    onOpenComplete: () => {},
-    onCloseComplete: () => {}
-  }
-
-  constructor(props, context) {
-    super(props, context)
-    this.state = initialState()
-  }
-
-  componentWillUnmount() {
-    if (this.latestAnimationFrame) {
-      cancelAnimationFrame(this.latestAnimationFrame)
+const Positioner = memo(forwardRef((props, ref) => {
+  const {
+    target,
+    isShown,
+    children,
+    initialScale = 0.9,
+    animationDuration = 300,
+    position = Position.BOTTOM,
+    bodyOffset = 6,
+    targetOffset = 6,
+    innerRef,
+    onOpenComplete = () => {},
+    onCloseComplete = () => {}
+  } = props
+
+  const [top, setTop] = useState(0)
+  const [left, setLeft] = useState(0)
```

Comment:
Should these be `null` to match the `initialState` from before?

Reply:
It could be, though it doesn't seem to have an effect (null vs 0). I like using 0, as its more descriptive of the values that will be set. Thoughts?